### PR TITLE
chore: require setting v2 image prefix in Makefile

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
To prevent accidentally pushing to the public images.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/v2/Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ SHELL = /bin/bash
 
 # create an .env file to override dev vars locally
 -include .env
-DEV_IMAGE_PREFIX?=gcr.io/ml-pipeline-test/dev/kfp-
+DEV_IMAGE_PREFIX?=gcr.io/dummy/dev/kfp-
```
Comment (gkcalat) [inline backend/src/v2/Makefile:20]: Is it intended to result in an error when at `docker push`? What if someone owns a GCR repository "dummy"?
Can we require defining a variable just before make: `GCR_REPO=gcr.io/myrepo make <rule>` and add `DEV_IMAGE_PREFIX="${GCR_REPO}/dev/kfp-"` inside the Makefile?

**Review by gkcalat — CHANGES_REQUESTED** [review]

**Comment by Linchin** [comment]
Just trying to think of a streamlined dev and release work flow to avoid issues like this in the future:
* Development: upload to `dummy` folder, keep using the `latest` label in master repo
* Release: 
    * Upload images to `ml-pipeline-test` folder
    * Tag the uploaded images with the corresponding version label that is going to be released
    * Replace the tag  `latest` with version label in the code where it downloads the images

I suggest:
* Rename `dummy` into `ml-pipeline-dev`
* Add the release process into the release code so it is automated

**Comment by chensun** [comment]
> Just trying to think of a streamlined dev and release work flow to avoid issues like this in the future:
> 
> * Development: upload to `dummy` folder, keep using the `latest` label in master repo
> * Release:
>   
>   * Upload images to `ml-pipeline-test` folder
>   * Tag the uploaded images with the corresponding version label that is going to be released
>   * Replace the path `dummy` with `ml-pipeline-test`
>   * Replace the tag  `latest` with version label in the code where it downloads the images
> 
> I suggest:
> 
> * Rename `dummy` into `ml-pipeline-dev`
> * Add the release process into the release code so it is automated.
> 
> In the future we may even add tags in the dev folder to avoid mix-ups.

In the development workflow, everyone should use their own project/repository, a shared one like `ml-pipeline-dev` won't work as multiple engineers may update it at the same time and interfere with each other. 
You're right that we should automate it in the release process, it's on our backlog before GA.

File: backend/src/v2/Makefile (hunk 1 above)
Comment (chensun) [reply backend/src/v2/Makefile]: Updated to result explicit error before building the image if the flag is not set.

Comment (gkcalat) [reply backend/src/v2/Makefile]: Thanks!

**Review by gkcalat — APPROVED** [review]

**Pull request merged**